buffer/src/test/java/io/netty/buffer/AbstractCompositeByteBufTest.java

```diff
@@ -1144,6 +1144,36 @@ public void testReleasesItsComponents() {
         assertEquals(0, buffer.refCnt());
     }
 
+    @Test
+    public void testReleasesItsComponents2() {
+        ByteBuf buffer = PooledByteBufAllocator.DEFAULT.buffer(); // 1
+
+        buffer.writeBytes(new byte[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10});
+
+        // use readRetainedSlice this time - produces different kind of slices
+        ByteBuf s1 = buffer.readRetainedSlice(2); // 2
+        ByteBuf s2 = s1.readRetainedSlice(2); // 3
+        ByteBuf s3 = s2.readRetainedSlice(2); // 4
+        ByteBuf s4 = s3.readRetainedSlice(2); // 5
+
+        ByteBuf composite = PooledByteBufAllocator.DEFAULT.compositeBuffer()
```

Comment:
same as above

Reply:
@normanmaurer Actually I don't think this one would make a difference since it's only used if/when the composite buffer gets expanded right?

Reply:
@njhill ah yeah true... then fix it please to use Unpooled.